Add onSetupError

PChambino opened the pull request:
Related to #8

jackfranklin commented on src/index.js:
```diff
@@ -56,29 +62,34 @@ const elmWebComponents = {
+          }
+        } catch (error) {
+          onSetupError(error)
+          throw error
```
I think we should only throw the error if the user hasn't passed an `onSetupError`. If I'd passed an `onSetupError` I think I'd be surprised to see the error also being thrown by the library. What do you think?

PChambino replied:
I understand your point. I added the `throw` so that during development something still gets logged to console since re-throwing the error shouldn't affect the application flow. Maybe using `console.error` would make the intention more clear. I will try `console.error` in development and push something to production to see if it behaves as expected.

jackfranklin commented:
Hey @PChambino this looks great! I have one thought about the error handling so let me know your thoughts, but very happy to merge this + publish once we've decided on that. Thank you :)

PChambino commented:
@jackfranklin 👋 Ok. I changed to `console.error` and come up with an extra thing. I added a `context` object to pass the flags to `onSetupError` since bad flags is one of the common errors of `ElmComponent.init` and pushing these flags together with the error to a bug tracking service should be helpful. (I didn't manage to push to production today, but will do first thing tomorrow.)

PChambino commented:
This has been running in production since this morning, everything working as expected. I am happy with this, let me know what you think.

jackfranklin commented:
@PChambino this is out as 1.0.0 now 👍
